### DISCUSSION TITLE
author_Insertkey Method not found

### DISCUSSION THREAD

**Discussion by comien** [ParityAsia/substrate-newsletter#15] [Substrate/Polkadot 技术问题] [answered]
![image](https://user-images.githubusercontent.com/45218301/108623301-41477900-7479-11eb-97ef-5c21e60d2863.png)
![image](https://user-images.githubusercontent.com/45218301/108623317-53291c00-7479-11eb-9f4f-643f7c8ad7f6.png)
请问一下，什么情况会发生Method not found？在自己电脑上的没问题，把代码挪到linux上编译启动，之后，使用author_insertKey就是Method not found，好像是命令里加了--rpc-external的原因，但是该怎么才能用呢？

**Comment by jimmychu0807** [comment]
这样没有上下文很难帮你。你有一少段代码给我们看可重现以上错误吗？

**Reply by comien** [reply]
```
./target/release/node-template --dev --tmp --ws-external --rpc-external --rpc-cors=all
```
在服务器上跑一个节点，想要通过polkadot.js.org/apps/，连接公网ip:9944，调用author_insertKey，就报错，Method Not Found

**Comment by jimmychu0807** [comment]
我用 [Substrate Node template v3.0.0](https://github.com/substrate-developer-hub/substrate-node-template/releases/tag/v3.0.0) 試了。可以呀。我只是跑 `--dev --tmp` 。

![Screenshot 2021-02-25 at 12 40 19](https://user-images.githubusercontent.com/898091/109105185-5b7aa300-7768-11eb-98f8-c1ca8f8fe6ea.png)

**Comment by comien** [comment, marked as answer]
我知道了，使用--ws-external，还要加上--rpc-methods=Unsafe，不然就出现Method Not found，🙏谢谢老师
